frontend/src/components/Calendar/Calendar.js

```diff
@@ -24,19 +39,92 @@ const Calendar = () => {
     setCurrentEvent(null);
   };
 
-  // Fetches events from API
+  // Function to handle states when an update form is opened
+  const openUpdateForm = () => {
+    setCurrentUpdate(currentEvent);
+    setIsUpdateOpen(true);
+    console.log("Opening update form for event:", currentEvent.title);
+  }
+
+  // Function to handle states when an update form is closed
+  const closeUpdateForm = () => {
+    console.log("Closing update form for event:", currentUpdate.title);
+    setCurrentUpdate(null);
+    setIsUpdateOpen(false);
+  }
+
+  // Function to live update (no browser refresh required) the calendar when an event is updated 
+  const updateEvent = (updatedEvent) => {
+    setEvents((previousEvents) => previousEvents.map((event) => event.id === updatedEvent.id ? updatedEvent : event));
+  };
+
+  // Function to handle states when a create form is opened
+  const openCreateForm = () => {
+    setIsCreateOpen(true);
+    console.log("Opening create form.");
+  }
+
+  // Function to handle states when a create form is closed
+  const closeCreateForm = () => {
+    setIsCreateOpen(false);
+    console.log("Closing create form");
+  } 
+
+  // Function to display a newly created event onto the calendar locally without refreshing the browser
+  const createEvent = (newEvent) => {
+    setEvents((prevEvents) => [...prevEvents, newEvent]); 
+    console.log("Created Event: ", newEvent.title);
+  }
+
+  // Function to delete the current event on a modal
+  const deleteEvent = async (deletedEvent) => {
+    try {
+      // Make DELETE request to delete the event
+      await api.delete(`${route}/${deletedEvent.id}/delete/`);
+      setEvents((previousEvents) => previousEvents.filter((event) => event.id !== deletedEvent.id));
+      closeModal();
+    } catch (error) {
+      console.log("Error deleting event: ", error.response);
+    } 
+  }
+
+  // Fetches events and hubs from API
   useEffect(() => {
-    console.log("Fetching...");
+
+    // Need to fix this. Works if first time trying to access website, but doesn't work for subsequent uses.
+    const token = localStorage.getItem(ACCESS_TOKEN);
```

Comment:
If this is something that you cannot fix currently, I think you should create a new issue to remind us to fix it in the future.

Reply:
I will remove it for now.